pallets/pallet-governance/src/tests.rs

```diff
@@ -0,0 +1,437 @@
+		// would not affect results.
+		assert_ok!(Governance::force_rotate_authorities(
+			RuntimeOrigin::root(),
+			(31..41).chain(31..35).collect::<Vec<_>>()
+		));
+
+		// Verify the authorities are set.
+		assert_eq!(CurrentAuthorities::<Runtime>::get(), (31..41).collect::<BTreeSet<_>>());
+
+		System::assert_last_event(RuntimeEvent::Governance(Event::<Runtime>::AuthorityRotated {
+			new_council: (31..41).collect::<BTreeSet<_>>(),
+		}));
+	});
+}
+
+#[test]
+fn can_end_to_end_council_rotate_authorities() {
+	MockGenesisConfig::with_authorities((11..21).collect::<Vec<_>>())
+		.build()
+		.execute_with(|| {
+			// Setup data
+			let call = Box::new(RuntimeCall::Governance(crate::Call::council_rotate_authorities {
+				new_members: (21..31).collect::<Vec<_>>(),
+			}));
+			let authority_member: AccountId = 11;
+			let first_proposal = 1u32;
+
+			// Propose
+			assert_ok!(Governance::initiate_proposal(
+				RuntimeOrigin::signed(authority_member),
+				call.clone()
+			));
+
+			// Vote
+			for i in 12..21 {
+				assert_ok!(Governance::vote(RuntimeOrigin::signed(i), first_proposal, true));
+			}
+
+			// dispatch
+			Governance::on_finalize(System::block_number());
+
+			// Verify the authorities are set.
+			assert_eq!(CurrentAuthorities::<Runtime>::get(), (21..31).collect::<BTreeSet<_>>());
+
+			assert_eq!(
+				Votes::<Runtime>::get(first_proposal),
+				CastedVotes { yays: Default::default(), nays: Default::default() }
+			);
+			assert_eq!(ProposalsToResolve::<Runtime>::get(), Default::default());
+			assert_eq!(Proposals::<Runtime>::get(first_proposal), None);
+
+			System::assert_has_event(RuntimeEvent::Governance(
+				Event::<Runtime>::AuthorityRotated {
+					new_council: (21..31).collect::<BTreeSet<_>>(),
+				},
+			));
+
+			let last_event = System::events().last().unwrap().event.clone();
+			matches!(last_event, RuntimeEvent::Governance(Event::<Runtime>::ProposalPassed {
+			id,
+			call,
+			result: Ok(..),
+		}) if id == first_proposal && call == call );
+		});
+}
+
+#[test]
+fn can_expired_council_rotate_authorities() {
+	MockGenesisConfig::with_authorities((11..21).collect::<Vec<_>>())
+		.build()
+		.execute_with(|| {
+			// Setup data
+			let call = Box::new(RuntimeCall::Governance(crate::Call::council_rotate_authorities {
+				new_members: (21..31).collect::<Vec<_>>(),
+			}));
+			let authority_member: AccountId = 11;
+			let first_proposal = 1u32;
+
+			// Propose
+			assert_ok!(Governance::initiate_proposal(
+				RuntimeOrigin::signed(authority_member),
+				call.clone()
+			));
+
+			// The number of votes is not enough to resolve the proposal.
+			for i in 12..15 {
+				assert_ok!(Governance::vote(RuntimeOrigin::signed(i), first_proposal, true));
+			}
+
+			// Expire the proposal.
+			Governance::on_finalize(System::block_number() + EXPIRY_PERIOD);
+
+			// Verify the authorities are original.
+			assert_eq!(CurrentAuthorities::<Runtime>::get(), (11..21).collect::<BTreeSet<_>>());
+
+			assert_eq!(
```

Comment:
also use contains key
You can write a helper function 
assert_storage_cleaned_up, which checks all 3 storages are cleared of a proposal ID. THis helps with duplicate codes